quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerImpl.java

```diff
@@ -30,4 +32,42 @@ public Collection addCollection(String name, String description) {
         return collectionRepo.save(collection);
     }
 
+    /**
+     * Finds a collection from mongodb collection table by _id
+     * 
+     * @param id used to look up the collection in mongodb
+     * 
+     * 
+     * @return Collection Instance that is found from the database
+     * 
+     **/
+    @Override
+    public Collection findCollection(String id) {
+        return collectionRepo.findBy_id(new ObjectId(id));
+    }
+
+    /**
+     * Edits an existing Collection and updates it in mongodb
+     * 
+     * @param _id of the collection that needs to be updated
+     * 
+     * 
+     * @return Collection Instance that is updated in database
+     * 
+     **/
+    @Override
+    public Collection editCollection(String id, Optional<String> name, Optional<String> description) {
+        Optional<Collection> collection = Optional.ofNullable(findCollection(id));
+        Collection updatedCollection = null;
+
+        if (collection.isPresent()) {
+            Collection currentCollection = collection.get();
+            name.ifPresent(updatedName -> currentCollection.setName(updatedName));
```

Comment:
FYI, I have added an Optional name and description, in case name and/or description are somehow passed as null, the app would still continue without a NullPointerException

Reply:
setting name to null shouldn't throw an exception (there is no operation done on `name` if you just call `setName(name)` and in some cases it might be desired.